pyxem/signals/segments.py

```diff
@@ -0,0 +1,444 @@
+                correlated_vector_intensities[i] = np.zeros(len(
+                    correlated_vector_indices[i]))
+                segment_mask = np.zeros_like(segment_mask)
+                segment_mask[np.where(correlated_segments[i])] = 1
+                segment_intensities = np.sum(self.segments.data * segment_mask,
+                                             axis=(1, 2))
+                for n, index in zip(range(len(correlated_vector_indices[i])),
+                                    correlated_vector_indices[i]):
+                    correlated_vector_intensities[i][n] = np.sum(
+                        segment_intensities[index])
+        else:
+            segment_intensities = np.sum(self.segments.data, axis=(1, 2))
+            for i in range(len(correlated_vectors)):
+                correlated_vector_intensities[i] = np.zeros(
+                    len(correlated_vector_indices[i]))
+                for n, index in zip(range(len(correlated_vector_indices[i])),
+                                    correlated_vector_indices[i]):
+                    correlated_vector_intensities[i][n] = np.sum(
+                        segment_intensities[index])
+
+        vdfseg = VDFSegment(Signal2D(correlated_segments),
+                            DiffractionVectors(correlated_vectors),
+                            correlated_vector_intensities)
+
+        # Transfer axes properties of segments
+        vdfseg.segments = transfer_signal_axes(vdfseg.segments, self.segments)
+        n = vdfseg.segments.axes_manager.navigation_axes[0]
+        n.name = 'n'
+        n.units = 'number'
+
+        return vdfseg
+
+    def get_virtual_electron_diffraction(self, calibration, shape, sigma=None):
+        """ Obtain a virtual electron diffraction signal that consists
+        of one virtual diffraction pattern for each segment. The virtual
+        diffraction pattern is composed of Gaussians centered at each
+        vector position. If given, the integrated intensities of each
+        vector will be taken into account by multiplication with the
+        Gaussians.
+
+        Parameters
+        ----------
+        calibration : float
+            Reciprocal space calibration in inverse Angstrom per pixel.
+        shape : tuple
+            Shape of the signal, (shape_x, shape_y) in pixels, where
+            shape_x and shape_y are integers.
+        sigma : float
+            The standard deviation of the Gaussians in inverse Angstrom
+            per pixel.
+
+        Returns
+        -------
+        virtual_ed : ElectronDiffraction2D
+            Virtual electron diffraction signal consisting of one
+            virtual diffraction pattern for each segment.
+        """
+        vectors = self.vectors_of_segments.data
+        segments = self.segments.data
+        num_segments = np.shape(segments)[0]
+
+        if self.intensities is None:
+            print("The VDFSegment does not have the attribute intensities."
```

Comment:
This should be a warning and needs a test.

Reply:
Will fix.

Reply:
fixed in https://github.com/dnjohnstone/pyxem/pull/4